### DISCUSSION TITLE
How to force ruff to process a directory called `build`?

### DISCUSSION THREAD

**Discussion by bjoernpollex** [astral-sh/ruff#14605] [Q&A]
I have a package in my python-project that's actually called `build` - is there a convenient way to make ruff process this specific package, without having to fully override `exclude` (which would require me to replicate the defaults)?

**Comment by MichaReiser** [comment]
No, I don't think there's an alternative to overriding `exclude` but I'm not the `exclude` defaults are the problem here because the defaults only exclude `_build` but not `build`. Any chance that `build` is excluded in a `gitignore` file?

**Reply by bjoernpollex** [reply]
Thanks for your reply! The `.gitignore` file contains the pattern `./build/`, which should only affect a `build`-directory at the repository root, right? In my case, the directory is not in root, but nested under a `src` folder, and the files in there are definitely managed by git. I'll play around a bit with manually overriding exclude.

**Reply by MichaReiser** [reply]
There's one known bug around gitignore https://github.com/astral-sh/ruff/issues/13692. It is quite possible that you're experiencing the same

**Reply by bjoernpollex** [reply]
Ok, apologies for wasting time - it seems I was simply on an outdated version of ruff - I've upgraded to 0.8.0 now and everything works fine!
